### ISSUE TITLE
Etsy OAuth2 PKCE missing x-api-key header or client_id

### ISSUE THREAD

**Issue by Rafurus** [n8n-io/n8n#7976]
**Describe the bug**
When trying to connect to an Etsy account using generic credential type from pkce, I get an error that the x-api-key or client_id is missing. 

**To Reproduce**

1. create an Etsy account
2. creating https://www.etsy.com/developers/your-apps application 
3. http request node configuration with OAuth2 PKCE
4. trying to connect the account

**Expected behavior**
Properly connect an account that will enable API calls.

**Environment (please complete the following information):**

- n8n Version 1.20

**Additional context**
1. Documentation -> https://developers.etsy.com/documentation/essentials/authentication 
2. Adding client_id=XXX in query parameters doesn't work.
3. Screen of error
![image](https://github.com/n8n-io/n8n/assets/12428720/3197bda7-197f-4016-9d1f-185de0ff9914)

**Comment by Joffcom** [comment]
Hey @Rafurus,

I am just waiting for Etsy to approve an application so I can give this a test, When you created your credential in n8n did you set the Authentication to Body or Header?

**Comment by Joffcom** [comment]
Hey @Rafurus,

I have a keystring and a shared secret but when trying to use the keystring as a client id it fails as the app doesn't exist.

**Comment by Rafurus** [comment]
@Joffcom 
Does your app being approved maybe?

**Comment by Joffcom** [comment]
Hey @Rafurus,

Still no approval 😞 I only have that keystring and shared secret which is no good. My worry here though is Etsy refer to it as a `specific implementation` so if they don't follow the normal standards we won't support it anyway and the best option would be a custom node / custom credential.

![image](https://github.com/n8n-io/n8n/assets/4688521/8191b6f5-e5ac-4a7a-bcd7-2cdda1263e4c)

**Comment by alexfacciorusso** [comment]
Have you solved this? I am having the exact same problem: the app can link and grants, but then when the hook is called, n8n throws that same error

**Comment by regedy1** [comment]
Anyone managed to connect successfully ? @Joffcom or @alexfacciorusso ?

**Comment by alexfacciorusso** [comment]
Nope. I think Etsy has a non-fully-standard implementation of oauth, or something similar, and the n8n auth component as per now doesn't allow for granular control over headers sent

**Comment by regedy1** [comment]
I got it to work with using Access token in header + x-api-key (etsy keystring.)

<img width="524" alt="Image" src="https://github.com/user-attachments/assets/a13d1708-65ee-4f65-8a34-a4cdf6940d6d" />

**Comment by Joffcom** [comment]
Etsy still have not approved my app however as this is a non standard implementation it would be an enhancement request more than a bug so I am going to mark this as closed for now.
